### ISSUE TITLE
Add support for PHP 7.4 arrow functions

### ISSUE THREAD

**Issue by jrfnl** [sirbrillig/phpcs-variable-analysis#134]
PHP 7.4 introduces arrow functions. It would be great if this library would add support for these.

```php
// Currently throwing false positives for undefined variables $a x 2, $b x 2, $c x 1.
$fn = fn($a, $b, $c) => $a + $b; // Unused $c (false negative).

function foo() {
    $a = 120;

	// Currently throwing false positive for undefined variable $b x 2.
    $fn1 = fn($b) => $a + $b;

	// Currently throwing false positive for undefined variable $b.
    $fn2 = fn($b) => $a; // Unused $b (false negative).

	// Currently throwing false positive for undefined variable $b x 2, $c x 1.
    $fn3 = fn($b) => $b + $c; // Undefined $c (false negative). Unused variable $fn3 (correctly thrown).
    
    var_dump( $fn1(10) );
    var_dump( $fn2(10) );
}

```

Note: PHPCS itself doesn't properly support arrow functions until PHPCS 3.5.5 (not yet released).
If continued support for older PHPCS versions is desired, you may want to look at the `PHPCSUtils` `FunctionDeclarations::isArrowFunction()` and `FunctionDeclarations::getArrowFunctionOpenClose()` methods.

Refs:
* https://wiki.php.net/rfc/arrow_functions_v2
* https://github.com/squizlabs/php_codesniffer/issues/2523

**Comment by sirbrillig** [comment]
Yes. Thanks for bringing this up. I'm very excited about PHP 7.4 but I hadn't started looking into it yet. No time like the present.

**Comment by speicus** [comment]
Subscribing for updates. Really hoping for a fix, unfortunately having to add `// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable` to each arrow function really discourages us from using them more widely.

**Comment by gertjankrol** [comment]
+1
